data/scripts.js

```diff
@@ -114,7 +114,10 @@ let BattleScripts = {
 			for (const dancer of dancers) {
 				if (this.faintMessages()) break;
 				this.add('-activate', dancer, 'ability: Dancer');
-				this.runMove(move.id, dancer, 0, this.getAbility('dancer'), undefined, true);
+				// @ts-ignore
+				const dancersTarget = target.side !== dancer.side && pokemon.side === dancer.side ? target : pokemon;
```

Comment:
```suggestion
				const dancersTarget = target && target.side !== dancer.side && pokemon.side === dancer.side ? target : pokemon;
```
If we pretend TypeScript is correct then we never want `target` to be falsy anyway, so with this you can remove both ts-ignores.

Reply:
See my further testing results below. I think it's pretty safe to assume that `target` being `null` in this code path is impossible, so being explicit with that by using a `ts-ignore` would be better practice in my opinion.

Reply:
That's fair, feel free to make pre's change above then.

<sup>_If this crashes one day I'll remember this!_ 😄